packages/desktop-client/src/components/reports/reports/CustomReportListCards.tsx

```diff
@@ -0,0 +1,191 @@
+// @ts-strict-ignore
+import React, { useState } from 'react';
+
+import { send } from 'loot-core/src/platform/client/fetch';
+
+import { styles } from '../../../style';
+import { Block } from '../../common/Block';
+import { Menu } from '../../common/Menu';
+import { MenuButton } from '../../common/MenuButton';
+import { MenuTooltip } from '../../common/MenuTooltip';
+import { View } from '../../common/View';
+import { ChooseGraph } from '../ChooseGraph';
+import { DateRange } from '../DateRange';
+import { LoadingIndicator } from '../LoadingIndicator';
+import { ReportCard } from '../ReportCard';
+
+type CardMenuProps = {
+  onClose: () => void;
+  onMenuSelect: (item: string, reportId: string) => void;
+  reportId: string;
+};
+
+function CardMenu({ onClose, onMenuSelect, reportId }: CardMenuProps) {
+  return (
+    <MenuTooltip onClose={onClose} width={120}>
+      <Menu
+        onMenuSelect={item => {
+          onMenuSelect(item, reportId);
+        }}
+        items={[
+          {
+            name: 'rename',
+            text: 'Rename report',
+            disabled: true,
+          },
+          {
+            name: 'delete',
+            text: 'Delete report',
+          },
+        ]}
+      />
+    </MenuTooltip>
+  );
+}
+
+function index(data) {
+  const result = {};
+  data.forEach(report => {
+    result[report.id] = false;
+  });
+  return result;
+}
+
+export function CustomReportListCards({ reports }) {
+  const result = index(reports);
+  const [reportMenu, setReportMenu] = useState(result);
+
+  const [isCardHovered, setIsCardHovered] = useState(null);
+
+  const onMenuSelect = async (item, reportId) => {
+    if (item === 'delete') {
+      onMenuOpen(reportId, false);
+      await send('report/delete', reportId);
+    }
+  };
+
+  const onMenuOpen = (item, state) => {
+    setReportMenu({ ...reportMenu, [item]: state });
+  };
+
+  const ReportGrid = slice => {
+    const pack = [];
+
+    slice?.map(item => {
+      pack.push({ data: item });
+      return null;
+    });
+
+    const data2 = [...pack];
+    const remainder = 3 - (data2.length % 3);
+    const groupedData = [];
+
+    while (data2.length) {
+      groupedData.push(data2.splice(0, 3));
+    }
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏this loop gets executed on every render. Please move it to `useMemo` + instead of mutating the original object you could use `slice`.